File: sorting/insertionSort/insertionSort.java
import java.util.Arrays;

public class insertionSort {

    static void insertionSortFn(int[] arr) {

        for (int i = 1; i < arr.length; i++) {
            int current = arr[i];
            int j;
            for (j = i - 1; j >= 0 && arr[j] > current; j--) {
                arr[j + 1] = arr[j];
            }
            if (j + 1 != i) {
                arr[j + 1] = current;
            }

        }
        System.out.println(Arrays.toString(arr));
    }

    public static void main(String[] args) {
        insertionSortFn(new int[] { 2, 1, 9, 76, 4 });
    }
}

File: namasteDSA/sorting/insertionSort.js
function insertionSort(arr){
    let n= arr.length;
    for(let i=1;i<n;i++){
        let current = arr[i];
        let prev = i-1;

        while(arr[prev]>current && prev>=0){
            arr[prev+1]=arr[prev];
            prev--
        }
        arr[prev+1] = current
    }
    return arr;
}
console.log(insertionSort([7,3,2,1,56,77,8,90]))